offchain/danaswap-api/src/DanaSwap/Api.purs

```diff
@@ -88,6 +94,31 @@ instance EncodeAeson Protocol where
       }
 
 type PoolId = TokenName
+type AssetClass = CurrencySymbol /\ TokenName
+
+newtype PoolDatum =
+  PoolDatum
+    { ac1 :: AssetClass
+    , ac2 :: AssetClass
+    , bal1 :: BigInt
+    , bal2 :: BigInt
+    , adminBal1 :: BigInt
+    , adminBal2 :: BigInt
```

Comment:
I'm assuming this is representing the admin balance for `a` and for `b`, right? If so, why not just use a tuple
```
adminBal :: (BigInt, BigInt)
```

Reply:
This bloats the datum